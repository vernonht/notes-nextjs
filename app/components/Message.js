import Image from "next/image";
import useStore from "@/store/store";
import parse from "html-react-parser";

export default function Message({ id, title, message }) {
  const currentNote = useStore((state) => state.currentNote);
  const isCurrentNote = currentNote === id;
  return (
    <div
      className={`
        ${
          isCurrentNote ? "bg-slate-600" : ""
        } flex items-center space-x-4 px-4 py-2 border-y border-gray-600 hover:bg-slate-600 hover:cursor-pointer transition-all
      `}
    >
      <div className="flex-none rounded-full h-14 w-14 overflow-hidden">
        <Image
          src="https://picsum.photos/56/56"
          width={56}
          height={56}
          alt="Picture of the author"
        />
      </div>
      <div className="grid">
        <div>{title}</div>
        <div className="text-sm text-gray-500 line-clamp-1">{parse(message)}</div>
      </div>
    </div>
  );
}
