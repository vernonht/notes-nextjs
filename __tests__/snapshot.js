import { render } from "@testing-library/react";
import Message from "../app/components/Message";

it("renders Message page unchanged", () => {
  const props = {
    id: 1,
    title: "Title",
    message:
      "Lorem Ipsum is simply dummy text of the printing and typesetting industry",
  };
  const { container } = render(
    <Message id={props.id} title={props.title} message={props.message} />
  );
  expect(container).toMatchSnapshot();
});
