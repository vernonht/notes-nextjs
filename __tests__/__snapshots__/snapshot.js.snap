// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Message page unchanged 1`] = `
<div>
  <div
    class="
         flex items-center space-x-4 px-4 py-2 border-y border-gray-600 hover:bg-slate-600 hover:cursor-pointer transition-all
      "
  >
    <div
      class="flex-none rounded-full h-14 w-14 overflow-hidden"
    >
      <img
        alt="Picture of the author"
        data-nimg="1"
        decoding="async"
        height="56"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fpicsum.photos%2F56%2F56&w=128&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fpicsum.photos%2F56%2F56&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fpicsum.photos%2F56%2F56&w=128&q=75 2x"
        style="color: transparent;"
        width="56"
      />
    </div>
    <div
      class="grid"
    >
      <div>
        Title
      </div>
      <div
        class="text-sm text-gray-500 line-clamp-1"
      >
        Lorem Ipsum is simply dummy text of the printing and typesetting industry
      </div>
    </div>
  </div>
</div>
`;
